cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1479,4 +1492,16 @@ private void resetLogLevels(HttpRequest request, HttpResponder responder, String
     }
     return result;
   }
+
+  private NamespaceId validateAndGetNamespace(String namespace) throws NamespaceNotFoundException {
+    NamespaceId namespaceId = Ids.namespace(namespace);
+    try {
+      namespaceQueryAdmin.get(namespaceId);
+    } catch (NamespaceNotFoundException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
```

Comment:
instead of this, just make this entire method throw Exception. We have too many usages of this method in our code and it makes proper error handling difficult, as in some places runtime exceptions are getting wrapped in other runtime exceptions.

Not sure why NamespaceQueryAdmin throws Exception and not just NamespaceNotFoundException...